src/PufferVaultMainnet.sol

```diff
@@ -80,20 +105,107 @@ contract PufferVaultMainnet is PufferVault {
         }
 
         uint256 assets = previewRedeem(shares);
+
+        _checkDailyWithdrawalLimits(assets); //@todo figure if it is better to override `maxWithdraw`
```

Comment:
doubt it's necessary to update maxWithdraw as it is global limit. Any edge cases you see it might be needed?